nmigen/vendor/xilinx_spartan_3_6.py

```diff
@@ -126,12 +135,28 @@ class XilinxSpartan6Platform(TemplatedPlatform):
         """,
         r"""
         {{get_tool("bitgen")}}
-            {{get_override("bitgen_opts")|default(["-w"])|options}}
+            {{get_override("bitgen_opts")|default(["-g Binary:Yes -w"])|options}}
             {{name}}_par.ncd
             {{name}}.bit
         """
     ]
 
+    @property
+    def family(self):
+        if self.device.startswith("xc3s"):
```

Comment:
This should normalize the case with something like `device = self.device.upcase()`.